src/block.js

```diff
@@ -191,19 +207,25 @@ const createUnsafe = ({ bytes, cid, value: maybeValue, codec }) => {
 
   if (value === undefined) throw new Error('Missing required argument, must either provide "value" or "codec"')
 
-  return new Block({ cid, bytes, value })
+  return new Block({
+    // eslint-disable-next-line object-shorthand
+    cid: /** @type {CID<T, Code, Alg, V>} */ (cid),
```

Comment:
What would this look like if it was written in TypeScript? Can you do an unsafe cast like this, or would you need to pass it through `CID.asCID()`, and should we be doing that here anyway?

Reply:
> What would this look like if it was written in TypeScript? 

```suggestion
    cid: <CID<T, Code, Alg, V>>cid
```
Not sure if it would require `cid:` or casting on shorthand would have worked

Reply:
> Can you do an unsafe cast like this, or would you need to pass it through CID.asCID(), and should we be doing that here anyway?

it is actually safe cast, if it wasn’t CID TS would not have allowed it. Here we just refine not fully known types unknown→ T, number → Code,   etc..


I think it’s right thing to do if we want to capture type parameters otherwise it would come out unrefined and reduce TS ability to do inference.